packages/carbon-component-mapper/src/files/text-field.js

```diff
@@ -7,22 +7,37 @@ import { TextInput, NumberInput } from 'carbon-components-react';
 import prepareProps from './prepare-props';
 
 const TextField = (props) => {
-  const { input, meta, validateOnMount, ...rest } = useFieldApi(prepareProps(props));
-
-  const Component = input.type === 'number' ? NumberInput : TextInput;
+  const { input, meta, validateOnMount, labelText, ...rest } = useFieldApi(prepareProps(props));
 
   const invalid = (meta.touched || validateOnMount) && (meta.error || meta.submitError);
   const warn = (meta.touched || validateOnMount) && meta.warning;
 
+  if (input.type === 'number') {
+    return (
+      <NumberInput
+        {...input}
+        key={input.name}
+        id={input.name}
+        invalid={Boolean(invalid)}
+        invalidText={invalid || ''}
+        warn={Boolean(warn)}
+        warnText={warn || ''}
+        label={labelText}
+        {...rest}
+      />
+    );
+  }
+
   return (
-    <Component
+    <TextInput
       {...input}
       key={input.name}
       id={input.name}
       invalid={Boolean(invalid)}
       invalidText={invalid || ''}
       warn={Boolean(warn)}
       warnText={warn || ''}
+      labelText={labelText}
```

Comment:
I like the previous approach more. :thinking: 

What about to just change the label/labelText:

```jsx
{...(type === 'number' ? { labelText } : { label: labelText } )}
```